src/wp-includes/block-template.php

```diff
@@ -242,8 +251,12 @@ function get_the_block_template_html() {
 	 * Even if the block template contained a `core/query` and `core/post-template` block referencing the main query
 	 * loop, it would not cause errors since it would use a cloned instance and go through the same loop of a single
 	 * post, within the actual main query loop.
+	 *
+	 * This special logic should be skipped if the current template does not come from the current theme, in which case
+	 * it has been injected by a plugin by hijacking the block template loader mechanism. In that case, entirely custom
+	 * logic may be applied which is unpredictable and therefore safer to omit this special handling on.
 	 */
-	if ( is_singular() && 1 === $wp_query->post_count && have_posts() ) {
+	if ( $current_theme_owns_current_template && is_singular() && 1 === $wp_query->post_count && have_posts() ) {
```

Comment:
```suggestion
	if (
		$_wp_current_template_id
		&& str_starts_with( $_wp_current_template_id, get_stylesheet() . '//' )
		&& is_singular()
		&& 1 === $wp_query->post_count
		&& have_posts()
	) {
```

Rather than adding a description in two separate places, could we implement a condition in a single location?